### ISSUE TITLE
--work-dir is gone and swagger generate spec from existing source is now broken

### ISSUE THREAD

**Issue by juliandroid** [go-swagger/go-swagger#2014]
## Problem statement
Checking out the master (latest commit 9b8be60e8a27f268afc3013da504b631cf623007 ) removes the --work-dir option and replace it with -b.
Using previous commit 9b8be60e8a27f268afc3013da504b631cf623007 the --work-dir is working properly.

Project structure is like:
```
project/
           cmd/
                  generate_swagger_cmd.sh
go.mod
```
Shell files contains:
swagger generate spec -w .. --scan-models -o ../../swagger-ui/dist/swagger.json

This command is run with "-w .." for perform generation of swagger.json file.

With the new version, the -w is gone and using -b instead produces empty json file
Getting rid entirely of "-b" generates incomplete json file - i.e. all ref: are not resolved.

## Environment
swagger version: commit 9b8be60e8a27f268afc3013da504b631cf623007 
go version go1.12.5 linux/amd64
go.mod based app

**Comment by casualjim** [comment]
You can change your command:

```swagger generate spec -m -o ../../swagger-ui/dist/swagger.json ../...```

**Comment by juliandroid** [comment]
Well, that is the problem:

If I do:
```bash
go get -u github.com/go-swagger/go-swagger/cmd/swagger

swagger generate spec --scan-models -o ../../swagger-ui/dist/swagger.json ../...
```
Then I get the following errors in the UI:
```
Resolver error at paths./media.post.responses.200.$ref
Could not resolve reference: Could not resolve pointer: /responses/uploadResp does not exist in document
Resolver error at paths./media.post.responses.400.$ref
Could not resolve reference: Could not resolve pointer: /responses/badReq does not exist in document
Resolver error at paths./media.post.responses.500.$ref
Could not resolve reference: Could not resolve pointer: /responses/internal does not exist in document
```

The problem is that all references like:

"$ref": "#/responses/uploadResp"  could not be resolved. In the swagger.json file fields like this one **are not generated/missing**:

```json
"uploadResp": {
      "description": "...",
      "schema": {
        "type": "object",
        "properties": {
          "code": {
            "description": "...",
            "type": "integer",
            "format": "int64",
          },
          "data": {
            "$ref": "#/definitions/..."
          }
        }
      }
    }
```

When using the old:

```bash
go get -u github.com/go-swagger/go-swagger/cmd/swagger@d7bdf7cb26243c4df640644efc313f64f7cd4256

swagger generate spec -w .. --scan-models -o ../../swagger-ui/dist/swagger.json
```

those references are there, so no missing fields and UI is happy.

Any suggestions how I could workaround that?

**Comment by casualjim** [comment]
could you give me a dummy layout of your files and how they are structured?
I have tests for this so it should show up, I just wonder if maybe you should be doing.

```swagger generate spec -m -o ../../swagger-ui/dist/swagger.json $(go list -m)/...```

**Comment by juliandroid** [comment]
```
$ tree
.
├── cmd
│   ├── arguments.go
│   ├── generate_swagger.sh
│   ├── logger.go
│   ├── server.go
│   ├── swagger
│   │   ├── media.go
│   │   └── model.go
│   ├── swaggerui
│   │   └── statik
│   │       └── statik.go
│   └── terminate.go
├── go.mod
├── go.sum
├── README.md
├── server
│   ├── media.go
│   ├── notimplemented.go
│   └── rest.go
├── upload
│   └── fs.go
└── uploader

```
For example, the respective missing uploadResp
```
// HTTP status code 200 and upload model in data
// swagger:response uploadResp
type swaggUploadResp struct {
```
is located at cmd/swagger/media.go (package swagger).
The main "swagger:meta" is located at cmd/server.go (package main)

**Comment by casualjim** [comment]
Yeah but in the new version we adhere to go's default scope selection
so  `-w .. -b .` and `../...` should be equivalent now. The loader will load all the code files from all the packages and all the files they import etc.

if you run with `DEBUG=1 swagger generate spec` it will print a lot of stuff but it may also tell us where it loses track.

This application is not public by any chance?

**Comment by juliandroid** [comment]
Yes, it seems that both are equivalent and they are working the same, but only if I use the old commit.
If I update to HEAD then spec generation stop working.

Please take a look at the debug output (ok.txt is using the old commit, while broken.txt is the latest commit from master). About source, sorry it is a private repo :(

[ok.txt](https://github.com/go-swagger/go-swagger/files/3388064/ok.txt)
[broken.txt](https://github.com/go-swagger/go-swagger/files/3388065/broken.txt)

**Comment by fredbi** [comment]
closed as no longer applicable
